src/header.iced

```diff
@@ -14,50 +13,60 @@ crypto_auth_KEYBYTES = 32
 crypto_secretbox_KEYBYTES = 32
 
 compute_mac_key = (encryptor, header_hash, pubkey) ->
-  zero_bytes = Buffer.alloc(crypto_auth_KEYBYTES)
+  zero_bytes = new Buffer(crypto_auth_KEYBYTES)
+  zero_bytes.fill(0)
   mac_box = encryptor.encrypt({plaintext : zero_bytes, nonce : nonce.nonceForMACKeyBox(header_hash), pubkey})
   # take last crypto_auth_BYTES bytes of MAC box
   return mac_box.slice(-crypto_auth_KEYBYTES)
 
-exports.generate_encryption_header_packet = (encryptor, recipients) ->
+exports.generate_encryption_header_packet = (encryptor, recipients, opts) ->
```

Comment:
My instinct is that both recipients and anoymous_recipients should be either both required (but allowed to be an empty list) or both optional. Either way we'll need a check to make sure that they're not both empty.